openproblems/tasks/spatial_decomposition/methods/nmfreg.py

```diff
@@ -0,0 +1,97 @@
+from ....tools.decorators import method
+from ....tools.normalize import log_cpm
+from ....tools.utils import check_version
+from sklearn.decomposition import NMF
+from sklearn.preprocessing import StandardScaler
+
+import collections
+import numpy as np
+import pandas as pd
+import scipy.optimize
+import scipy.stats
+
+
+@method(
+    method_name="NMF-reg",
+    paper_name="Slide-seq: A scalable technology for measuring genome-wide expression at high spatial resolution",
+    paper_url="https://science.sciencemag.org/content/363/6434/1463",
+    paper_year=2009,
+    code_url="N/A",
+    code_version=check_version("NMF-reg"),
+)
+def nmfreg(adata):
+    n_types = adata.obsm["proportions_true"].size[1]
+    adata_sc = adata.uns["sc_reference"].copy()
+    factors = 30
+    projectiob_type = "l2"
+
+    # Learn from reference
+    X = adata_sc.X.toarray()
```

Comment:
just wondering if it is necessary for this to be sparse ? Prob yes if the `scikit-learn.NMF` doesn't support sparse, but would double check just in case.

Reply:
Good point, I converted to dense for later operations, like standard-scalar, normalization, row operations. NMF works with sparse.